libs/frontend/feature-build/src/components/build-render/ProseMirrorEditor.vue

```diff
@@ -25,20 +34,66 @@ const mountProseMirrorEditor = (
 ) => {
   const { selectedComponent } = editor ?? {}
   if (editor && container && selectedComponent) {
+    const plugins: Plugin[] = []
+
+    // Pass gradient color styles to the inner div of ProseMirror editor.
+    plugins.push(
+      new Plugin({
+        props: {
+          attributes: { style: gradientColorStyle.value },
+        },
+      }),
+    )
+
     editor.editView?.destroy()
     editor.editView = createComponentEditorView(
       {
         content: selectedComponent.content ?? '',
+        plugins,
       },
       container,
     )
   }
 }
 
+const gradientColorStyle = computed(() => {
```

Comment:
This seems like a pretty complex change just to get gradients working. Will it apply more generally if we also need to pass other styles to the prosemirror editor?

Reply:
Oh, I was just about to leave some comments to describe what I've changed, but you reviewed it so fast that I'm not even done 😅.

I accidentally found out a couple minutes ago that if we remove the `position: relative` at `pubstudio-builder/libs/frontend/ui-theme/src/css/components/prosemirror-view.postcss:4`, the text gradient problem would be magically solved!
I'm not very sure why ProseMirror set that style; I guess it's because ProseMirror editor is allowed to display HTML content, the `position: relative` is to make sure that absolute positioning elements will not be at somewhere outside the editor.

If we plan to allow users to paste HTML content to the editor in the future, maybe we should keep this `position: relative`. But that means we'll have to think of a way to create PubStudio components based on the pasted HTML content.

Reply:
Ah, sorry about that! I was going through all the PRs just now.

So do you prefer to revert this change and remove `position: relative` instead? I don't really understand why that fixes it though, and I worry a bit that it will be a fragile solution.

Long term, my idea is that a subset of html can be copy/pasted into the editor, maybe only `span/h/a/p`, but I'm not sure yet. ProseMirror would still control the html/layout, but we would have a plugin to allow styling text selections based on editor commands. I think the styles would be inlined in that case.

Reply:
I think the current solution (passing styles from component to ProseMirror editor) is more reliable, because I also don't understand why removing `position: relative` fixes the bug 😅! I'm not sure if it's something related to `-webkit-text-fill-color` and/or `-webkit-background-clip`.

From [this sandbox](https://stackblitz.com/edit/web-platform-gfdp58?file=index.html) it seems that text gradient can be applied to the descendant no matter how deep it is. But adding `position: relative` in the middle does make the text invisible, just like what we've seen in the ProseMirror editor. So I think it's a HTML thing rather than a ProseMirror-specific issue.

Reply:
Interesting! Let's stick with your solution.

Although I just noticed something strange, which may not be fixable:

![Screenshot 2023-11-10 at 3 57 10 PM](https://github.com/samatechtw/pubstudio-builder/assets/550982/67eee7db-80a6-4041-b998-0f4d3e1efced)

The gradient is applied over all the text, rather than each individual line. I suppose it would be really rare to actually want to have multiple lines of gradient text though 🤔